gwdetchar/lasso/plot.py

```diff
@@ -72,3 +73,67 @@ def save_figure(fig, pngfile, **kwargs):
             return
     fig.close()
     return pngfile
+
+
+def make_spectrum_plot(unfiltered_spectrum, filtered_spectrum, channel_name,
+                       x_min, x_max, y_min, y_max, x_label, y_label, title,
+                       filename):
+    """Generate and return a spectrum plot
+    """
+    fig = unfiltered_spectrum.plot(label='Unfiltered', color='#ee0000')
+    ax = fig.gca()
+    ax.plot(filtered_spectrum, label='Filtered', color='#92ddc8')
+    ax.set_xlim(x_min, x_max)
+    ax.set_ylim(y_min, y_max)
+    ax.set_xlabel('Frequency [Hz]', fontsize='small')
+    ax.set_ylabel(y_label, fontsize='small')
+    ax.set_title(title, fontsize='small')
+    ax.legend(loc='best', fontsize='small')
+    spectrum_plot = save_figure(
+        fig, filename, bbox_inches='tight')
+    fig.close()
+    return spectrum_plot
+
+
+def make_spectrum_plots(start, end, flower, fupper, channel_name,
+                        unfiltered_spectrum, filtered_spectrum):
+    """Generate and return both spectrum plots
+    """
+    re_delim = re.compile(r'[:_-]')
+
+    # Attributes shared between zoomed out and zoomed in plots
+    channelstub = re_delim.sub('_', str(channel_name)).replace('_', '-', 1)
+    gpsstub = '%d-%d' % (start, end-start)
+    x_label = 'Frequency [Hz]'
+    if ':GDS-CALIB_STRAIN' in channel_name:
+        y_label = 'GW amplitude spectral density [strain/$\sqrt{\mathrm{Hz}}]$'
+    else:
+        y_label = 'Primary channel units'
+    y_min = 1e-24
+    y_max = 2e-18
```

Comment:
These y axis limits are great for when strain is the primary channel, but would they need to be decided in a more general way for other possible primary channels?

Reply:
@macedo22, I see what you mean, how about something like this?

```python
    y_min = 0.1 * filtered_spectrum.min().value
    y_max = 5 * filtered_spectrum.max().value
```

I think we can live with this plot always showing up with a log scaled y-axis, but this change would make the axis limits more adaptive.